js/ui/windowManager.js

```diff
@@ -1832,3 +1887,608 @@ const WindowManager = new Lang.Class({
         }
     },
 });
+
+const ICON_BOUNCE_MAX_SCALE = 0.4;
+const ICON_BOUNCE_ANIMATION_TIME = 1.0;
+const ICON_BOUNCE_ANIMATION_TYPE_1 = 'easeOutSine';
+const ICON_BOUNCE_ANIMATION_TYPE_2 = 'easeOutBounce';
+
+const BUTTON_OFFSET_X = 50;
+const BUTTON_OFFSET_Y = 50;
+
+function animateBounce(actor) {
+    Tweener.removeTweens(actor);
+    if (Tweener.isTweening(actor))
+        return;
+
+    Tweener.addTween(actor, {
+        scale_y: 1 + ICON_BOUNCE_MAX_SCALE,
+        scale_x: 1 + ICON_BOUNCE_MAX_SCALE,
+        translation_y: actor.height * ICON_BOUNCE_MAX_SCALE,
+        translation_x: actor.width * ICON_BOUNCE_MAX_SCALE / 2,
+        time: ICON_BOUNCE_ANIMATION_TIME * 0.25,
+        delay: 0.3,
+        transition: ICON_BOUNCE_ANIMATION_TYPE_1
+    });
+    Tweener.addTween(actor, {
+        scale_y: 1,
+        scale_x: 1,
+        translation_y: 0,
+        translation_x: 0,
+        time: ICON_BOUNCE_ANIMATION_TIME * 0.35,
+        transition: ICON_BOUNCE_ANIMATION_TYPE_2,
+        delay: ICON_BOUNCE_ANIMATION_TIME * 0.25 + 0.3,
+        onComplete: function() {animateBounce(actor);}
+    });
+}
+
+const CodingManager = new Lang.Class({
+    Name: 'CodingManager',
+
+    _init: function(actor) {
+        this._sessions = [];
+        this._rotateInActors = [];
+        this._rotateOutActors = [];
+        this._firstFrameConnections = [];
+        this._previousFocusedWindow = null;
```

Comment:
The this._previousFocusedWindow is unused indeed, removed it. The rest is still in use.